pkg/user/decode.go

```diff
@@ -0,0 +1,44 @@
+package user
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/gorilla/mux"
+)
+
+var ErrBadRouting = errors.New("inconsistent mapping between route and handler (programmer error)")
```

Comment:
Should you use the error package?

Reply:
yes I got it to a tefactor enhance of project